### PR TITLE
Check round trip before doing cocina update or create

### PR DESCRIPTION
**Pull request by justinlittman**
## Why was this change made?
To make sure that only changes that can be roundtripped are accepted for create or update. This ensures the integrity of data that is submitted via cocina update or create endpoints.

## How was this change tested?
Unit

## Which documentation and/or configurations were updated?
NA

### CODE REVIEW

File: app/services/cocina/object_creator.rb [hunk 1]
```diff
@@ -155,5 +154,15 @@ def apply_default_access(item)
     def truncate_label(label)
       label.length > 254 ? label[0, 254] : label
     end
+
+    def validate(obj)
+      if Settings.enabled_features.validate_descriptive_roundtrip.create
+        validator = DescriptionRoundtripValidator.new(obj)
+        raise ValidationError.new(validator.error, status: :internal_server_error) unless validator.valid?
```
Comment (jcoyne) [inline app/services/cocina/object_creator.rb:161]: I'm not sure that 500 is the right code here. Would you consider 409?

File: app/controllers/objects_controller.rb [hunk 2]
```diff
@@ -39,6 +42,9 @@ def update
 
     render json: cocina_object
   rescue Cocina::ValidationError => e
+    # Notify if this is a roundtripping error.
+    Honeybadger.notify(e) if e.status == :internal_server_error
```
Comment (jcoyne) [inline app/controllers/objects_controller.rb:46]: Can this be a subclass of `Cocina::ValidationError` so we don't have to test on properties?

File: app/services/cocina/object_updater.rb [hunk 3]
```diff
@@ -144,7 +145,13 @@ def validate
       raise ValidationError, validator.error unless validator.valid?
 
       raise NotImplemented, 'Updating descriptive metadata not supported' if !update_descriptive? && client_attempted_metadata_update?
+
+      if has_changed?(:description) && update_descriptive? && Settings.enabled_features.validate_descriptive_roundtrip.update
```
Comment (jcoyne) [inline app/services/cocina/object_updater.rb:149]: why not do a `return unless ...` here?

File: app/validators/cocina/description_roundtrip_validator.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+      true
+    end
+
+    private
+
+    attr_reader :cocina_object
+
+    def druid
+      # Requests do not have druids.
```
Comment (jcoyne) [inline app/validators/cocina/description_roundtrip_validator.rb:35]: 👍

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: app/services/cocina/object_creator.rb (hunk 1 above)
Comment (justinlittman) [reply app/services/cocina/object_creator.rb]: Done.

File: app/controllers/objects_controller.rb (hunk 2 above)
Comment (justinlittman) [reply app/controllers/objects_controller.rb]: Done.

File: app/services/cocina/object_updater.rb (hunk 3 above)
Comment (justinlittman) [reply app/services/cocina/object_updater.rb:149]: While I generally support guard statements, I find this sort of a use of a guard statement confusing and requires additional effort to understand the code. In particular, I think guard statements should be used to filter the unhappy/atypical path, leaving the normal code path as the happy/typical path. In this case, if filters the happy/typical path and leaves the normal code path the unhappy/atypical path.

Comment (justinlittman) [reply app/services/cocina/object_updater.rb:149]: But this is a pet peeve of mine.

**Review by jcoyne — APPROVED** [review]

**Pull request merged**